From 87d3c6a000febacfaf5dd8af29062166eee328a2 Mon Sep 17 00:00:00 2001
From: Mehmet Killioglu <mehmet.killioglu@unikie.com>
Date: Thu, 1 Jun 2023 12:18:00 +0300
Subject: [PATCH] [PATCH] CMakeLists.txt: making package relocatable

---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8089906..7220a7e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,14 +21,15 @@ target_include_directories(laser_geometry
   PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:include/${PROJECT_NAME}>
-  ${Eigen3_INCLUDE_DIRS}
 )
 ament_target_dependencies(laser_geometry
   "rclcpp"
   "sensor_msgs"
   "tf2"
 )
-
+target_link_libraries(laser_geometry
+    Eigen3::Eigen
+)
 # Causes the visibility macros to use dllexport rather than dllimport,
 # which is appropriate when building the dll but not consuming it.
 target_compile_definitions(laser_geometry PRIVATE "LASER_GEOMETRY_BUILDING_LIBRARY")
-- 
2.34.1

